README.md

```diff
@@ -8,11 +8,45 @@ An example on creating a k3k cluster on an RKE2 host using k3kcli
 
 [![asciicast](https://asciinema.org/a/eYlc3dsL2pfP2B50i3Ea8MJJp.svg)](https://asciinema.org/a/eYlc3dsL2pfP2B50i3Ea8MJJp)
 
-## Usage
+## Architecture
 
 K3K consists of a controller and a cli tool, the controller can be deployed via a helm chart and the cli can be downloaded from the releases page.
 
-### Deploy Controller
+### Controller
+
+The K3K controller will watch a CRD called `clusters.k3k.io` Once found, the controller will create a separate namespace and it will create a K3S cluster as specified in the spec of the object.
+
+Each server and agent is created as a separate pod that runs in that namespace.
+
+### CLI
+
+The CLI will provide a quick way to create K3K clusters using simple flags. The CLI automatically exposes the K3K clusters so it's accessible via a kubeconfig.
+
+## Features
+
+### Isolation
+
+Each cluster runs in a sperate namespace that can be isolated via netowrk policies and RBAC rules, clusters also run in a sperate network namespace with flannel as the backend CNI, finally each cluster has a separate datastore which can be persisted.
```

Comment:
```suggestion
Each cluster runs in a sperate namespace that can be isolated via netowrk policies and RBAC rules, clusters also run in a sperate network namespace with flannel as the backend CNI. Finally, each cluster has a separate datastore which can be persisted.
```

Reply:
Can we add a bit here about how the datastore can be persisted? I think there'd be value in highlighting how users can have the benefit of Kine, etc.

Reply:
I think it still needs some code modifications, since we are defaulting now to the etcd backend, it can be done but I opted not to mention it until its actually available